library/lvm_gensym.py

```diff
@@ -0,0 +1,152 @@
+author: 
+    - Tim Flannagan (tflannag@redhat.com)
+'''
+
+EXAMPLES = '''
+- name: Generate names 
+  lvm_gensym:
+    fs_type: "{{ fs_type }}"
+    mount: "{{ mount_point }}"
+  register: lvm_results
+  when: lvm_vg == "" and mount_point != "" and fs_type != ""
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+'''
+
+RETURN = '''
+vg_name:
+    description: The default generated name for an unspecified volume group
+    type: str 
+
+lv_name:
+    description: The default generated name for an unspecified logical volume 
+    type: str
+'''
+
+def get_os_name():    
+    for line in open('/etc/os-release').readlines():
+        if not line.find('ID='):
+            os_name = line[3:]
+            break
+        
+    os_name = os_name.replace('\n', '').replace('"', '')
+    return os_name 
+
+def name_is_unique(name, used_names):
+    if name not in used_names:
+        return True 
+    
+    return False 
+
+def get_unique_name_from_base(base_name, used_names):
+    counter = 0
+    while not name_is_unique(base_name, used_names):
+        if counter == 0:
+            base_name = base_name + '_' + str(counter)
+        else:
+            base_name = base_name[:-2] + '_' + str(counter)
+        counter += 1
+
+    return base_name 
+
+def get_vg_name_base(host_name, os_name):
+    if host_name != None and len(host_name) != 0:
+        vg_default = os_name + '_' + host_name
+    else:
+        vg_default = os_name + '_default' 
+
+    return vg_default
+
+def get_vg_name(host_name, os_name, lvm_facts):
+    used_vg_names = lvm_facts['vgs'].keys()
+    used_vg_names = [elem.lower() for elem in used_vg_names]
+    name = get_vg_name_base(host_name, os_name)
+
+    return get_unique_name_from_base(name, used_vg_names)    
+
+def get_lv_name_base(fs_type, mount_point):
+    if 'swap' in fs_type.lower():
+        lv_default = 'swap'
+    elif mount_point.startswith('/'):
+        if mount_point == '/':
+            lv_default = 'root'
+        else:
+            lv_default = mount_point[1:].replace('/', '_')
+    else:
+        lv_default = 'lv'
+
+    return lv_default 
+    
+def get_lv_name(fs_type, mount_point, lvm_facts):
+    used_lv_names = lvm_facts['lvs'].keys()
+    used_lv_names = [elem.lower() for elem in used_lv_names]
+    name = get_lv_name_base(fs_type, mount_point)
+
+    return get_unique_name_from_base(name, used_lv_names)    
+
+def run_module():
+    module_args = dict(
+        mount=dict(type='str', required=True),
+        fs_type=dict(type='str', required=True)
+    )
+
+    result = dict(
+        changed=True,
+        vg_name='',
+        lv_name=''
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True
+    )
+
+    lvm_facts = facts.ansible_facts(module)['lvm']
+    host_name = facts.ansible_facts(module)['nodename'].lower().replace('.', '_').replace('-', '_')
+    os_name   = get_os_name()
```

Comment:
Since you don't use `os_name` otherwise in this function, you could instead call `get_os_name` from `get_vg_name`, allowing you to drop one of the arguments to `get_vg_name`.